src/app/locales/vi_VN.ts

```diff
@@ -0,0 +1,513 @@
+
+const uiLabels: LocalizedLabels = {
+  'Angular Update Guide': 'Hướng dẫn Update Angular',
+  'Select the options matching your project:': 'Tuỳ chỉnh những lựa chọn cho phù hợp với dự án của bạn:',
+  'Angular Versions': 'Phiên bản Angular',
+  From: 'Từ',
+  To: 'Đến',
+  Warning: 'Chú ý',
+  'Plans for releases after the current major release are not finalized and may change. These recommendations are based on scheduled deprecations.':
+    'Đối với những phiên bản Angular được phát hành sau phiên bản chính thức hiện tại, kế hoạch và dự định cho những phiên bản này chưa được thông qua chính thức và có thể sẽ thay đổi. Những gợi ý sau dựa trên những chỉnh sửa đã được thống nhất.',
+  'We do not currently support downgrading versions of Angular.':
+    'Chúng tôi hiện tại không hỗ trợ Downgrade phiên bản Angular',
+  'We do not recommend moving across multiple major versions.':
+    'Chúng tôi không khuyến khích update Angular theo kiểu "nhảy cóc" phiên bản.',
+  'App Complexity': 'Độ phức tạp của ứng dụng',
+  Basic: 'Cơ bản',
+  Medium: 'Vừa',
+  Advanced: 'Nâng cao',
+  'Show update information relevant to all Angular developers.':
+    'Hiển thị thông tin update dành cho tất cả mọi tầng lớp lập trình viên Angular.',
+  'Show more specific update information relevant to around 10% of Angular developers.':
+    'Hiển thị thông tin update chi tiết hơn dành cho khoảng 10% lập trình viên Angular. ',
+  'Show all the information we have about the update.': 'Hiển thị tất cả thông tin về việc update mà chúng tôi biết được.',
+  'Other Dependencies': 'Những yêu cầu liên quan khác',
+  'I use': 'Tôi sử dụng',
+  'Package Manager': 'Package Manager',
+  npm: 'npm',
+  yarn: 'yarn',
+  'Show me how to update!': 'Hướng dẫn tôi update!',
+  'Before Updating': 'Trước khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any changes needed before moving between these versions.':
+    'Hiện tại thì bạn không cần phải chuẩn bị gì trước khi update giữa những phiên bản này.',
+  'During the Update': 'Trong khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any recommendations for moving between these versions.':
+    'Hiện tại thì chúng tôi không có những gợi ý gì cho bạn cho việc update giữa những phiên bản này.',
+  'After the Update': 'Sau khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any changes needed after moving between these versions.':
+    'Hiện tại thì bạn không cần phải làm thêm gì sau khi update giữa những phiên bản này.',
+};
+
+const steps: LocalizedSteps = {
+  'Extends OnInit': {
+    action:
+      'Bảo đảm là bạn không dùng `extends OnInit`, hoặc dùng `extends` đối với bất kỳ LifeCycle nào. Thay vào đó, dùng `implements <LifeCycle>',
+  },
+  'Deep Imports': {
+    action:
+      'Dừng việc sử dụng Deep Imports, những biểu tượng này đều được đánh dấu với ký tự `ɵ` và không nằm trong API được public ra ngoài.',
```

Comment:
public may be confusing, use "API công chúng"?

Reply:
I disagree. I don't even know if I understand a guide/article that refers to Public API as "API công chúng".